src/plugin_output_processing/provider.py

```diff
@@ -18,60 +18,77 @@
 
 import ollama
 import openai
+from httpx import ConnectError
 from loguru import logger
 
+OLLAMA_NAME = "ollama"
+OPENAI_NAME = "openai"
+
 
 class ProviderError(Exception):
     pass
 
 
 class Provider:
-
     name: str
-    models: list[str] = []
-    available: bool
+    available: bool = False
     url: str = None
-
-    def get_model(self, model) -> str:
-        if not model:
-            logger.warning(f"Model not set for {self.name}. Using default.")
-            model = self.default
-        if model not in self.models:
-            logger.warning(f"Model {model} not found in {self.name} models.")
-            model = self.default
+    model: str
+
+    def get_model(
+        self,
+        list_models: list[str],
+        model: str | None = None,
+        default_model: str | None = None,
+    ) -> str:
+        if default_model is None:
+            default_model = list_models[0]
+        if model is None:
+            return default_model
+        if model not in list_models:
+            logger.warning(f"{model} not found in model list, using default instead.")
+            return default_model
         return model
```

Comment:
For me it is not necessary clearer with class attributes especially because those class attributes are only use in one method. But if it is more clear for you, then I don't mind.

I commited some changes in that direction